src/core/features/login/pages/credentials/credentials.ts

```diff
@@ -137,6 +143,10 @@ export class CoreLoginCredentialsPage implements OnInit, OnDestroy {
                 }
             });
         }
+
+        this.alwaysShowLoginFormObserver = CoreEvents.on<number>(ALWAYS_SHOW_LOGIN_FORM_CHANGED, async () => {
```

Comment:
```suggestion
        this.alwaysShowLoginFormObserver = CoreEvents.on(ALWAYS_SHOW_LOGIN_FORM_CHANGED, async () => {
```

To define the parameters of an event we do it differently so it's detected automatically without having to specify the type in every listener. You should add the type in here:

https://github.com/moodlehq/moodleapp/blob/main/src/core/features/login/services/login-helper.ts#L1695

In this code we're basically adding new values to the `CoreEventsData` interface that is used by `CoreEvents`. The keys of the object are the names of the events, and the value is the type of the parameter passed to the event. If you add your event in here, all the listeners will automatically infer the type without having to specify it manually. So you can remove the `<number>` type from here and from reconnect page.

PD: Please notice the type of the parameter you're passing is `{ value: number }`, not just `number`.